Bio/Align/__init__.py

```diff
@@ -2202,8 +2211,16 @@ def format(self, fmt="", *args, **kwargs):
                        the alignment and include it in the output. If False
                        (default), do not include the MD tag in the output.
         """
+        substitution_matrix = None
+        writer_argument = None
+        if isinstance(argument, PairwiseAligner):
+            substitution_matrix = argument.substitution_matrix
+        elif isinstance(argument, (np.ndarray, substitution_matrices.Array)):
+            substitution_matrix = argument
+        else:
+            writer_argument = argument
```

Comment:
I haven't tried it, but I guess a separate `writer_argument` won't be needed if you pull `argument` from `kwargs`